File: Data Structures & Algorithms/two-integer-sum/submission-3.java

class Solution {
    // import java.util.HashMap;
    public int[] twoSum(int[] nums, int target) {
        //Store num (key) and index (value)
        HashMap<Integer,Integer> numsMap = new HashMap<>();
        int[] returnVal = {0,0};
        numsMap.put(nums[0], 0);
        for(int i = 1; i <= nums.length - 1; i++){

            int dif = target - nums[i];
            if(numsMap.containsKey(dif)){
                if (i > numsMap.get(dif)){
                    returnVal[0] = numsMap.get(dif);
                    returnVal[1] = i;
                    return returnVal;
                }
                else{
                    returnVal[0] = i;
                    returnVal[1] = numsMap.get(dif);
                    return returnVal;
                }
            }
            else{
                numsMap.put(nums[i], i);
            }
        }
        return returnVal;
    }
}

File: Data Structures & Algorithms/two-integer-sum-ii/submission-3.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        if len(numbers) == 2:
            return [1,2]
        l = 0
        r = len(numbers) - 1
        sum = numbers[l] + numbers[r]
        while(sum != target):
            if(sum > target):
                r -= 1
            else:
                l += 1
            sum = numbers[l] + numbers[r]
        ans = [l+1,r+1]
        return ans
        
